### ISSUE TITLE
Mrustc 0.11.0's rustc-1.74.0-src.patch does not apply cleanly

### ISSUE THREAD

**Issue by dtolnay** [thepowersgang/mrustc#349]
```console
$ git clone https://github.com/thepowersgang/mrustc --branch=v0.11.0
$ cd mrustc
$ export RUSTC_VERSION=1.74.0 MRUSTC_TARGET_VER=1.74 
$ make RUSTCSRC
```

Output on Ubuntu 24.04 (GNU patch 2.7.6):

```console
make -f minicargo.mk RUSTCSRC
make[1]: Entering directory '/git/tmp/repro/mrustc'
[CURL] rustc-1.74.0-src.tar.gz
tar -xf rustc-1.74.0-src.tar.gz
touch rustc-1.74.0-src/extracted
cd rustc-1.74.0-src/ && patch -p0 < ../rustc-1.74.0-src.patch;
patching file compiler/rustc_parse_format/src/lib.rs
patching file compiler/rustc_ast/src/ast.rs
patching file compiler/rustc_hir/src/hir.rs
patching file compiler/rustc_errors/src/lib.rs
patching file compiler/rustc_const_eval/src/interpret/operand.rs
patching file compiler/rustc_const_eval/src/interpret/place.rs
patching file compiler/rustc_index/src/bit_set.rs
patching file library/std/src/sys_common/wtf8.rs
patching file compiler/rustc_ast/src/tokenstream.rs
patching file compiler/rustc_middle/src/mir/consts.rs
patching file compiler/rustc_middle/src/mir/interpret/value.rs
patching file compiler/rustc_middle/src/mir/syntax.rs
patching file compiler/rustc_middle/src/mir/mod.rs
patching file compiler/rustc_middle/src/ty/consts/kind.rs
patching file vendor/hex/src/lib.rs
patch unexpectedly ends in middle of line
Hunk #1 succeeded at 239 with fuzz 1.
touch rustc-1.74.0-src//dl-version
make[1]: Leaving directory '/git/tmp/repro/mrustc'
```

Output on macOS 15.1.1 (patch 2.0-12u11-Apple):

```console
/Library/Developer/CommandLineTools/usr/bin/make -f minicargo.mk RUSTCSRC
[CURL] rustc-1.74.0-src.tar.gz
tar -xf rustc-1.74.0-src.tar.gz
touch rustc-1.74.0-src/extracted
cd rustc-1.74.0-src/ && patch -p0 < ../rustc-1.74.0-src.patch;
patching file 'compiler/rustc_parse_format/src/lib.rs'
patching file 'compiler/rustc_ast/src/ast.rs'
patching file 'compiler/rustc_hir/src/hir.rs'
patching file 'compiler/rustc_errors/src/lib.rs'
patching file 'compiler/rustc_const_eval/src/interpret/operand.rs'
patching file 'compiler/rustc_const_eval/src/interpret/place.rs'
patching file 'compiler/rustc_index/src/bit_set.rs'
patching file 'library/std/src/sys_common/wtf8.rs'
patching file 'compiler/rustc_ast/src/tokenstream.rs'
patching file 'compiler/rustc_middle/src/mir/consts.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/mir/consts.rs.rej'
patching file 'compiler/rustc_middle/src/mir/interpret/value.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/mir/interpret/value.rs.rej'
patching file 'compiler/rustc_middle/src/mir/syntax.rs'
patching file 'compiler/rustc_middle/src/mir/mod.rs'
patching file 'compiler/rustc_middle/src/ty/consts/kind.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/ty/consts/kind.rs.rej'
patching file 'vendor/hex/src/lib.rs'
1 out of 1 hunks failed--saving rejects to 'vendor/hex/src/lib.rs.rej'
make[1]: *** [rustc-1.74.0-src//dl-version] Error 1
make: *** [RUSTCSRC] Error 2
```

The content of the .rej files, in the same order as the log above:

```patch
@@ -73,2 +73,2 @@
-#[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
+#[cfg(all(target_arch = "x86_64", target_pointer_width = "64", not(rust_compiler = "mrustc")))]
 static_assert_size!(ConstValue<'_>, 24);
```

```patch
@@ -37,2 +37,2 @@
-#[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
+#[cfg(all(target_arch = "x86_64", target_pointer_width = "64", not(rust_compiler = "mrustc")))]
 static_assert_size!(Scalar, 24);
```

```patch
@@ -75,2 +75,2 @@
-#[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
+#[cfg(all(target_arch = "x86_64", target_pointer_width = "64", not(rust_compiler = "mrustc")))]
 static_assert_size!(Expr<'_>, 24);
```

```patch
@@ -239,2 +239,2 @@
-#[cfg(target_pointer_width = "64")]
+#[cfg(all(target_pointer_width = "64", not(rust_compiler = "mrustc")))]
 from_hex_array_impl! {
\ No newline at end of line
```

**Comment by thepowersgang** [comment]
Odd, it works in CI (and locally)... although, I've seen issues attempting with patch not realizing that the files have already been patched.

Most of those patches are hand-crafted, so maybe there's something subtly wrong with them...

**Comment by dtolnay** [comment]
Probably the Apple `patch` wants to see at least 1 line of context before and after the lines changed. The hunks that failed to apply are exactly the ones without any context.

**Comment by thepowersgang** [comment]
That could be it, silly apple

**Comment by thepowersgang** [comment]
Hopefully fixed in bb8b59f16c3cd234f4f08102b57558748766dec0

**Comment by dtolnay** [comment]
Confirmed fixed. Thank you.

```console
/Library/Developer/CommandLineTools/usr/bin/make -f minicargo.mk RUSTCSRC
[CURL] rustc-1.74.0-src.tar.gz
tar -xf rustc-1.74.0-src.tar.gz
touch rustc-1.74.0-src/extracted
cd rustc-1.74.0-src/ && patch -p0 < ../rustc-1.74.0-src.patch;
patching file 'compiler/rustc_parse_format/src/lib.rs'
patching file 'compiler/rustc_ast/src/ast.rs'
patching file 'compiler/rustc_hir/src/hir.rs'
patching file 'compiler/rustc_errors/src/lib.rs'
patching file 'compiler/rustc_const_eval/src/interpret/operand.rs'
patching file 'compiler/rustc_const_eval/src/interpret/place.rs'
patching file 'compiler/rustc_index/src/bit_set.rs'
patching file 'library/std/src/sys_common/wtf8.rs'
patching file 'compiler/rustc_ast/src/tokenstream.rs'
patching file 'compiler/rustc_middle/src/mir/consts.rs'
patching file 'compiler/rustc_middle/src/mir/interpret/value.rs'
patching file 'compiler/rustc_middle/src/mir/syntax.rs'
patching file 'compiler/rustc_middle/src/mir/mod.rs'
patching file 'compiler/rustc_middle/src/ty/consts/kind.rs'
patching file 'vendor/hex/src/lib.rs'
touch rustc-1.74.0-src//dl-version
```
